server/graphql/v1/mutations/orders.js

```diff
@@ -510,36 +504,98 @@ export async function createOrder(order, loaders, remoteUser, reqIp) {
       collective.addUserWithRole({ id: user.id, CollectiveId: order.ReferralCollectiveId }, roles.FUNDRAISER);
     }
 
-    // Share suspicious transactions on Slack
-    if (recaptchaResponse && recaptchaResponse.score && recaptchaResponse.score <= 0.5) {
-      slackLib
-        .postActivityOnPublicChannel(
-          {
-            type: activities.ORDERS_SUSPICIOUS,
-            data: {
-              order,
-              user,
-              fromCollective,
-              collective,
-              recaptchaResponse,
-            },
-          },
-          config.slack.webhookUrl,
-          {
-            channel: config.slack.abuseChannel,
-          },
-        )
-        .catch(console.log);
+    return order;
+  } catch (error) {
+    // console.log(error);
+
+    if (orderCreated) {
+      if (!orderCreated.processedAt) {
+        if (error.stripeResponse) {
+          orderCreated.status = status.PENDING;
+        } else {
+          orderCreated.status = status.ERROR;
+        }
+        orderCreated.data.error = { message: error.message };
+        orderCreated.save();
+      }
+
+      if (!error.stripeResponse) {
+        throw error;
+      }
+
+      orderCreated.stripeError = {
+        message: error.message,
+        account: error.stripeAccount,
+        response: error.stripeResponse,
+      };
+
+      return orderCreated;
+    }
+
+    throw error;
+  }
+}
+
+export async function confirmOrder(order, remoteUser) {
+  if (!remoteUser) {
+    throw new errors.Unauthorized({ message: 'You need to be logged in to confirm an order' });
+  }
+
+  order = await models.Order.findOne({
+    where: {
+      id: order.id,
+    },
+    include: [
+      { model: models.Collective, as: 'collective' },
+      { model: models.Collective, as: 'fromCollective' },
+      { model: models.PaymentMethod, as: 'paymentMethod' },
+      { model: models.Subscription, as: 'Subscription' },
+    ],
+  });
+
+  if (!order) {
+    throw new errors.NotFound({ message: 'Order not found' });
+  }
+  if (!remoteUser.isAdmin(order.FromCollectiveId)) {
```

Comment:
Tricky case: if order is made with a gift card, to one that will need to confirm it is the gift card emitter - not the person ordering. In such case we'll need to check `PaymentMethod.CollectiveId`